recirq/qaoa/placement.py

```diff
@@ -41,6 +37,35 @@ def calibration_data_to_graph(calib_dict: Dict) -> nx.Graph:
     return err_graph
 
 
+def _qubit_index_edges(device, qubit_to_index):
+    """Helper function in `_device_to_tket_device`"""
+    dev_graph = ccr.xmon_device_to_graph(device)
+    for n1, n2 in dev_graph.edges:
+        yield qubit_to_index[n1], qubit_to_index[n2]
+
+
+def _device_to_tket_device(device, qubit_to_index):
+    """Custom function to turn a device into a pytket device.
+
+    Note: there's a problem if you try to use tket's 2D grid-qubit-like
+    Node objects, so we use the `qubit_to_index` mapping to use contiguous
+    integer Node ids.
+
+    This supports any device that supports `ccr.xmon_device_to_graph`.
+    """
+    arc = pytket.routing.Architecture(
+        list(_qubit_index_edges(device, qubit_to_index))
+    )
+    return pytket.device.Device({}, {}, arc)
+
+
+def _tk_to_i(tk):
+    """Extract an integer from pytket nodes."""
+    i = tk.index
+    assert len(i) == 1, i
```

Comment:
Just `assert len(i) == 1`?

Reply:
In the off chance that it isn't 1, it's nice to have the actual value immediately printed in the exception to jump-start debugging